backend/workflow_manager/workflow/workflow_helper.py

```diff
@@ -112,30 +114,37 @@ def process_input_files(
         destination: DestinationConnector,
         execution_service: WorkflowExecutionServiceHelper,
         single_step: bool,
-        hash_values_of_files: dict[str, str] = {},
+        hash_values_of_files: dict[str, FileHash] = {},
     ) -> WorkflowExecution:
-        input_files = source.list_files_from_source()
-        total_files = len(input_files)
+        input_files, total_files = source.list_files_from_source(hash_values_of_files)
         error_message = None
         processed_files = 0
         error_raised = 0
         execution_service.publish_initial_workflow_logs(total_files)
         execution_service.update_execution(
             ExecutionStatus.EXECUTING, increment_attempt=True
         )
-        for index, input_file in enumerate(input_files):
+        if total_files > 0:
+            q_file_no_list = WorkflowUtil.get_q_no_list(workflow, total_files)
```

Comment:
@vishnuszipstack what does this do? Is it specific to MRQ? Will it be a source of regression for other types?

Reply:
@chandrasekharan-zipstack this is specific for MRQ